src/lambda/commands/deploySamApplication.ts

```diff
@@ -40,7 +43,17 @@ export async function deploySamApplication({
         const buildTemplatePath = path.join(buildDestination, 'template.yaml')
         const outputTemplatePath = path.join(tempFolder, 'template.yaml')
         let stage: 'starting up' | 'building' | 'packaging' | 'deploying' = 'starting up'
+
+        const settingsConfiguration = new DefaultSettingsConfiguration(extensionSettingsPrefix)
+        const profile = settingsConfiguration.readSetting<string>(profileSettingKey)
```

Comment:
Consider making a "Get Current Credential name" method so that we have something reusable, and to curb this method's linecount growth. #maintenance 

(Comment applies to lines 52-56 also)

Reply:
As a general utility or just a part of this function? I don't see much of a reason inside this file, and moving it to a utility method could make it more difficult to mock the context (see @mpiroc 's comment)